File: Examples/c/07_arrays.c
/*
 * Arrays - Working with Collections
 * 
 * Demonstrates:
 * - Array declaration and initialization
 * - Accessing array elements
 * - Iterating through arrays
 * - Multi-dimensional arrays
 * - Common array operations
 */

#include <stdio.h>

int main() {
    int i, j;
    
    // ARRAY BASICS
    printf("=== Array Basics ===\n");
    int numbers[5] = {10, 20, 30, 40, 50};
    
    printf("Array elements:\n");
    for (i = 0; i < 5; i++) {
        printf("numbers[%d] = %d\n", i, numbers[i]);
    }
    printf("\n");
    
    // ARRAY OPERATIONS
    printf("=== Array Operations ===\n");
    
    // Sum
    int sum = 0;
    for (i = 0; i < 5; i++) {
        sum += numbers[i];
    }
    printf("Sum: %d\n", sum);
    
    // Average
    float average = (float)sum / 5;
    printf("Average: %.1f\n", average);
    
    // Find maximum
    int max = numbers[0];
    for (i = 1; i < 5; i++) {
        if (numbers[i] > max) {
            max = numbers[i];
        }
    }
    printf("Maximum: %d\n\n", max);
    
    // STORING SCORES
    printf("=== Student Scores ===\n");
    int scores[10];
    
    printf("Enter 10 test scores:\n");
    for (i = 0; i < 10; i++) {
        printf("Score %d: ", i + 1);
        scanf("%d", &scores[i]);
    }
    
    // Calculate average
    sum = 0;
    for (i = 0; i < 10; i++) {
        sum += scores[i];
    }
    printf("\nClass average: %.1f\n\n", (float)sum / 10);
    
    // 2D ARRAY - Matrix
    printf("=== 2D Array: Multiplication Table ===\n");
    int table[5][5];
    
    // Fill table
    for (i = 0; i < 5; i++) {
        for (j = 0; j < 5; j++) {
            table[i][j] = (i + 1) * (j + 1);
        }
    }
    
    // Print table
    printf("     1   2   3   4   5\n");
    printf("   -------------------\n");
    for (i = 0; i < 5; i++) {
        printf("%d | ", i + 1);
        for (j = 0; j < 5; j++) {
            printf("%3d ", table[i][j]);
        }
        printf("\n");
    }
    
    return 0;
}

File: Examples/pascal/07_arrays.pas
{
  Arrays - Working with Collections
  
  Demonstrates:
  - Array declarations
  - Accessing array elements
  - Iterating through arrays
  - Multi-dimensional arrays
}

Program Arrays;
Var
  numbers: Array[1..5] Of Integer;
  scores: Array[1..10] Of Integer;
  table: Array[1..5, 1..5] Of Integer;
  i, j, sum, max: Integer;
  average: Real;

Begin
  { ARRAY BASICS }
  WriteLn('=== Array Basics ===');
  numbers[1] := 10;
  numbers[2] := 20;
  numbers[3] := 30;
  numbers[4] := 40;
  numbers[5] := 50;
  
  WriteLn('Array elements:');
  For i := 1 To 5 Do
    WriteLn('numbers[', i, '] = ', numbers[i]);
  WriteLn;
  
  { ARRAY OPERATIONS }
  WriteLn('=== Array Operations ===');
  
  { Sum }
  sum := 0;
  For i := 1 To 5 Do
    sum := sum + numbers[i];
  WriteLn('Sum: ', sum);
  
  { Average }
  average := sum / 5.0;
  WriteLn('Average: ', average:0:1);
  
  { Find maximum }
  max := numbers[1];
  For i := 2 To 5 Do
    If numbers[i] > max Then
      max := numbers[i];
  WriteLn('Maximum: ', max);
  WriteLn;
  
  { STORING SCORES }
  WriteLn('=== Student Scores ===');
  WriteLn('Enter 10 test scores:');
  For i := 1 To 10 Do
  Begin
    Write('Score ', i, ': ');
    ReadLn(scores[i]);
  End;
  
  { Calculate average }
  sum := 0;
  For i := 1 To 10 Do
    sum := sum + scores[i];
  WriteLn;
  WriteLn('Class average: ', (sum / 10.0):0:1);
  WriteLn;
  
  { 2D ARRAY - Multiplication Table }
  WriteLn('=== 2D Array: Multiplication Table ===');
  
  { Fill table }
  For i := 1 To 5 Do
    For j := 1 To 5 Do
      table[i, j] := i * j;
  
  { Print table }
  Write('     ');
  For j := 1 To 5 Do
    Write(j:3, ' ');
  WriteLn;
  WriteLn('   -------------------');
  
  For i := 1 To 5 Do
  Begin
    Write(i, ' | ');
    For j := 1 To 5 Do
      Write(table[i, j]:3, ' ');
    WriteLn;
  End;
End.
